msbuild/Xamarin.iOS.Tasks.Windows/Xamarin.iOS.Common.After.targets

```diff
@@ -0,0 +1,406 @@
+		differ by size too.  Instead, this target uses inputs and outputs to only copy when the source is newer.
+		-->
+		<Copy Condition="'$(IsMacEnabled)' == 'true'"
+			SessionId="$(BuildSessionId)"
+			SourceFiles="@(_ResolvedFileToPublishPreserveNewest)"
+			DestinationFiles="@(_ResolvedFileToPublishPreserveNewest->'$(PublishDir)%(RelativePath)')"
+			OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
+			Retries="$(CopyRetryCount)"
+			RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
+			UseHardlinksIfPossible="$(CreateHardLinksForPublishFilesIfPossible)"
+			UseSymboliclinksIfPossible="$(CreateSymbolicLinksForPublishFilesIfPossible)">
+
+			<Output TaskParameter="DestinationFiles" ItemName="FileWrites"/>
+
+		</Copy>
+
+	</Target>
+
+	<!-- This overrides the NET SDK target, it does the same thing but we add the SessionId property so the copy is also executed remotely -->
+	<!-- https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L252-L279 -->
+	<Target Name="_CopyResolvedFilesToPublishAlways" DependsOnTargets="_ComputeResolvedFilesToPublishTypes">
+
+		<!--
+		Use SkipUnchangedFiles to prevent unnecessary file copies. The copy will occur if the
+		destination doesn't exist, the source is newer than the destination, or if the source and
+		destination differ by file size.
+		-->
+		<Copy Condition="'$(IsMacEnabled)' == 'true'"
+			SessionId="$(BuildSessionId)"
+			SourceFiles = "@(_ResolvedFileToPublishAlways)"
+			DestinationFiles="@(_ResolvedFileToPublishAlways->'$(PublishDir)%(RelativePath)')"
+			SkipUnchangedFiles="$(SkipCopyUnchangedFiles)"
+			OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
+			Retries="$(CopyRetryCount)"
+			RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
+			UseHardlinksIfPossible="$(CreateHardLinksForPublishFilesIfPossible)"
+			UseSymboliclinksIfPossible="$(CreateSymbolicLinksForPublishFilesIfPossible)">
+
+			<Output TaskParameter="DestinationFiles" ItemName="FileWrites"/>
+
+		</Copy>
+	</Target>
+
+	<PropertyGroup>
+		<_UpdateDynamicLibraryIdDependsOn>
+			$(_UpdateDynamicLibraryIdDependsOn);
+			CopyDynamicLibrariesToMac;
+		</_UpdateDynamicLibraryIdDependsOn>
+	</PropertyGroup>
+
+	<Target Name="CopyDynamicLibrariesToMac" Condition="'$(_PlatformName)' != 'macOS'">
+		<ItemGroup>
+			<_MonoLibraryFixToMac Include="%(_MonoLibraryFix.FullPath)">
+				<TargetPath>%(FullPath)</TargetPath>
+			</_MonoLibraryFixToMac>
+		</ItemGroup>
+
+		<CopyFilesToBuildServer 
+			Condition="'$(IsMacEnabled)' == 'true'" 
+			SessionId="$(BuildSessionId)" 
+			Files="@(_MonoLibraryFixToMac)" />
+	</Target>
+
+	<!-- Overrides Core SDK target to remote the ILLink execution -->
+	<!-- https://github.com/dotnet/sdk/blob/cdf57465e1cba9e44b5c9a76a403d41b1b8f178c/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L76-L132 -->
+	<Target Name="_RunILLink"
```

Comment:
We should come up with some supported way of doing this, because _RunILLink task is under active development and changes pretty frequently from what I understand (so this code will pretty soon be out-of-date). I don't think that should block this PR though.